app/coffee/WebsocketController.coffee

```diff
@@ -101,6 +101,29 @@ module.exports = WebsocketController =
 							logger.err {err, project_id, doc_id, fromVersion, line, client_id: client.id}, "error encoding line uri component"
 							return callback(err)
 						escapedLines.push line
+					if options.encodeRanges
+						if ranges.comments
+							escapedComments = []
+							for comment in ranges.comments
+								try
+									comment.op.c = unescape(encodeURIComponent(comment.op.c))
+								catch err
+									logger.err {err, project_id, doc_id, fromVersion, comment, client_id: client.id}, "error encoding comment uri component"
+									return callback(err)
+								escapedComments.push comment
+							ranges.comments = escapedComments
+						if ranges.changes
+							escapedChanges = []
+							for change in ranges.changes
+								try
+									change.op.i = unescape(encodeURIComponent(change.op.i)) if change.op.i
```

Comment:
I think it's worth abstracting the `unescape(encodeURIComponent( ... ))` into an 'encodeForWebsockets` method now we use it in multiple places.